### PR TITLE
Coach report tests

### LINKED ISSUE #3707: Coach reports - Tests!
Create tests for the whole thing. This is really a few separate tasks:

- [x] Writing features and implementing steps for the behave tests.
- [x] Deleting or modifying existing tests.
- [x] Creating new unit tests where appropriate.

### PR DESCRIPTION
**Pull request by rtibbles**
Fixes #3707

Summary of changes:
* Implements steps for Coach Report BDD tests
* Deletes old tests
* Updates generaterealdata management command to create attemptlogs

### CODE REVIEW

**Comment by rtibbles** [comment]
Also fixes #3799

File: kalite/coachreports/api_views.py [hunk 1]
```diff
@@ -154,9 +154,9 @@ def aggregate_learner_logs(request):
     output_dict["learner_events"] = [{
         "learner": log.user.get_name(),
         "complete": log.complete,
-        "struggling": log.struggling,
-        "progress": log.streak_progress,
-        "exercise": get_exercise_cache().get(log.exercise_id),
+        "struggling": getattr(log, "struggling", None),
+        "progress": getattr(log, "streak_progress", getattr(log, "progress", None)),
+        "content": get_exercise_cache().get(getattr(log, "exercise_id"), get_content_cache().get(getattr(log, "video_id", getattr(log, "content_id")), {})),
```
Comment (MCGallaspy) [inline kalite/coachreports/api_views.py]: Is your intent that the value here depends on the log type? It's easier for my puny human eyes to parse something that explicitly depends e.g. `type(log)` than a chain of defaults.

Comment (rtibbles) [inline kalite/coachreports/api_views.py]: Yes, that is indeed my intent.

If it were not being built inside a list comprehension, I would try to be a little less succinct. Also happy to comment it a bit if that's also helpful.

Comment (MCGallaspy) [inline kalite/coachreports/api_views.py]: Yeah, I see what you mean. One way would be to create an intermediary mapping in the comprehension (e.g. `"type": mycoollogtype`, and then apply a `map` to the resulting generator (or whatever it is), but it's a bit clumsy. A comment would probably do nicely.

File: kalite/coachreports/features/coachreports.feature [hunk 2]
```diff
@@ -1,81 +1,70 @@
...
+        And there should be three learner rows displayed
+        And there should be ten exercise columns displayed
+        And I should see a Hide Tabular Report button
+
+    Scenario Outline: I want to read student data from the tabular report
+        Given the "<learner>" "<progress verbs>" an "<exercise>"
+        And I am on the tabular report
+        Then I should see the "<exercise>" marked for "<learner>" as "<progress text>" and coloured "<progress colour>"
+
+    Examples: Learners
```
Comment (MCGallaspy) [inline kalite/coachreports/features/coachreports.feature]: Fancy.

**Comment by MCGallaspy** [comment]
The upshot here is that the only failing tests are seemingly unrelated and should be fixed in other PRs. A quick review and I'll be ready to merge this.

File: kalite/testing/behave_helpers.py [hunk 3]
```diff
@@ -169,6 +185,16 @@ def find_xpath_with_wait(context, id_str, **kwargs):
     """
     return _find_elem_with_wait(context, (By.XPATH, id_str), **kwargs)
 
+def find_css_with_wait(context, id_str, **kwargs):
+    """ Tries to find an element with given XPATH with an explicit timeout.
```
Comment (MCGallaspy) [inline kalite/testing/behave_helpers.py]: > Tries to find an element with given XPATH

No it doesn't!

File: kalite/coachreports/features/steps/coachreports.py [hunk 4]
```diff
@@ -17,5 +36,181 @@ def step_impl(context):
...
+    warning = find_css_class_with_wait(context, "alert-warning")
+    assert warning, "No warning shown."
+
+@then(u"I should be taken to the learner's progress report page")
+def impl(context):
+    assert reverse("student_view") in context.browser.current_url, "Assertion failed. {url} not in {current_url}".format(url=reverse("student_view"), current_url=context.browser.current_url)
+
+@given(u"I am on the detail panel")
+def impl(context):
+    context.execute_steps(u"""
```
Comment (MCGallaspy) [inline kalite/coachreports/features/steps/coachreports.py]: Someone's been reading the docs!

File: kalite/coachreports/features/steps/coachreports.py [hunk 5]
```diff
@@ -17,5 +36,181 @@ def step_impl(context):
...
+    user = FacilityUser.objects.get(username=learner)
+    data_row = find_id_with_wait(context, user.id)
+    cell = data_row.find_element_by_css_selector("td[value={exercise}]".format(exercise=exercise))
+    assert cell.text == progress_text, "Progress text for {learner}, on {exercise} incorrect".format(learner=learner, exercise=exercise)
+    assert rgba_to_hex(cell.value_of_css_property("background-color")) == colour_legend[progress_colour]
+
+@when(u"I click on the learner name")
+def impl(context):
+    student_name = find_css_class_with_wait(context, "student-name")
+    student_name.find_element_by_tag_name("a").click()
```
Comment (MCGallaspy) [inline kalite/coachreports/features/steps/coachreports.py]: Is the click here supposed to trigger a page load? Then you should use [click_and_wait_for_page_load](https://github.com/learningequality/ka-lite/blob/develop/kalite/testing/behave_helpers.py#L56).

File: kalite/coachreports/features/steps/coachreports.py [hunk 6]
```diff
@@ -17,5 +36,181 @@ def step_impl(context):
...
+    assert len(context.browser.find_elements_by_css_selector(".headrow.data")) == 10
+
+@when(u"I click on the dropdown button under the Group label")
+def impl(context):
+    find_id_with_wait(context, "group-select").click()
+
+@when(u"I click on the exercise name")
+def impl(context):
+    headrow = find_css_class_with_wait(context, "headrow")
+    headrow.find_element_by_tag_name("a").click()
```
Comment (MCGallaspy) [inline kalite/coachreports/features/steps/coachreports.py]: Same deal; does this trigger a page load?

**Pull request closed without merging**